build.psm1

```diff
@@ -679,22 +679,77 @@ function Start-PSxUnit {
 }
 
 
+function Install-Dotnet {
+    [CmdletBinding()]
+    param(
+        [string]$Channel = "rel-1.0.0",
+        [string]$Version = "latest",
+        [switch]$NoSudo
+    )
+
+    $obtainUrl = "https://raw.githubusercontent.com/dotnet/cli/rel/1.0.0/scripts/obtain"
+
+    # Install for Linux and OS X
+    if ($IsLinux -or $IsOSX) {
+        # Uninstall all previous dotnet packages
+        $uninstallScript = if ($IsUbuntu) {
+            "dotnet-uninstall-debian-packages.sh"
+        } elseif ($IsOSX) {
+            "dotnet-uninstall-pkgs.sh"
+        }
+
+        if ($uninstallScript) {
+            Start-NativeExecution {
+                curl -sO $obtainUrl/uninstall/$uninstallScript
+                Invoke-Expression "$sudo bash ./$uninstallScript"
```

Comment:
Oh I meant to copy the line back up for `$sudo`. Dynamic scoping is the reason it worked anyway; I don't like that.

A helper function may be the better answer. Can you come up with one? Sergei and I tried for a bit but found only `Invoke-Expression` to be happy taking an optional `sudo` prepend.